TypeScript types are broken under Node native ESM

## 🐛 Bug Report

When using `type-check` with TypeScript in a Node.js package with `"type": "module"` (native ESM), the types do not resolve.

## To Reproduce

Steps to reproduce:

- Clone https://github.com/ezzatron/fast-check-types-repro
- Run `npm ci`
- Run `npx tsc --noEmit -p tsconfig.json`
- Observe the following errors:

```
$ npx tsc --noEmit -p tsconfig.json
example.ts:1:10 - error TS2614: Module '"fast-check"' has no exported member 'double'. Did you mean to use 'import double from "fast-check"' instead?

1 import { double } from "fast-check"
           ~~~~~~

node_modules/fast-check/lib/esm/types/fast-check.d.ts:1:21 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

1 import * as fc from './fast-check-default';
                      ~~~~~~~~~~~~~~~~~~~~~~

node_modules/fast-check/lib/esm/types/fast-check.d.ts:3:15 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

3 export * from './fast-check-default';
                ~~~~~~~~~~~~~~~~~~~~~~


Found 3 errors in 2 files.

Errors  Files
     1  example.ts:1
     2  node_modules/fast-check/lib/esm/types/fast-check.d.ts:1
```

You can also see this output in GitHub Actions: https://github.com/ezzatron/fast-check-types-repro/actions/runs/6289635563/job/17076317889#step:5:5

## Expected behavior

No TypeScript errors when using `"moduleResolution": "nodeNext"`. See https://www.typescriptlang.org/docs/handbook/esm-node.html

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check           |    `3.13.0`        |
| node                 |     `20.7.0`       |
| TypeScript\*         |      `5.2.2`      |

\*Only for TypeScript's users

--- a/.yarn/versions/fdc3fd4c.yml
+++ b/.yarn/versions/fdc3fd4c.yml
@@ -1,0 +1,8 @@
+releases:
+  fast-check: patch
+
+declined:
+  - "@fast-check/ava"
+  - "@fast-check/jest"
+  - "@fast-check/vitest"
+  - "@fast-check/worker"

--- a/packages/fast-check/postbuild/main.cjs
+++ b/packages/fast-check/postbuild/main.cjs
@@ -12,7 +12,7 @@ const replace = require('replace-in-file');
 // Append *.js file extension on all local imports
 
 const options = {
-  files: 'lib/**/*.js',
+  files: ['lib/**/*.js', 'lib/**/*.d.ts'],
   from: [/from '\.(.*)(?<!\.js)'/g, /from "\.(.*)(?<!\.js)"/g],
   to: ["from '.$1.js'", 'from ".$1.js"'],
 };

